packages/ses/src/module-load.js

```diff
@@ -72,18 +77,24 @@ const loadRecord = async (
     resolvedImports,
   });
 
+  for (const fullSpecifier of values(resolvedImports)) {
+    // Behold: recursion.
+    // eslint-disable-next-line no-use-before-define
+    const loadDependency = memoizedLoadWithErrorAnnotation(
+      compartmentPrivateFields,
+      moduleAliases,
+      compartment,
+      fullSpecifier,
+      queue,
+      loading,
+    );
+    // Queuing does not work well with unhandled rejection warnings.
+    promiseCatch(loadDependency, noop);
+    setAdd(queue, loadDependency);
+  }
+
   // Memoize.
   mapSet(moduleRecords, moduleSpecifier, moduleRecord);
-
-  // Await all dependencies to load, recursively.
-  await promiseAll(
```

Comment:
This is the crux of the change. I took this await Promise.all out and converted it into a queue that must be fully awaited by the end of the call to `load` that spawns all the work. That allows every edge of the import graph to settle independently and avoids cyclic recursion.